### PR TITLE
feat(post-scan action): Create terraform version for the aws-promote-and-quarantine-plugin

### PR DESCRIPTION
**Pull request by felipecosta09**
# Add terraform template option

## Change Summary

This update will allow deployment of the promote-and-quarantine plugin using terraform

## PR Checklist

- [X] I've read and followed the [Contributing Guide](https://github.com/trendmicro/cloudone-filestorage-plugins/blob/master/.github/CONTRIBUTING.md).
- Documents/Readmes
  - [X] Updated accordingly
  - [ ] Not required
- Plugins that have versioning
  - [ ] Version bumped and follows [Semantic Versioning](https://semver.org/)
  - [X] Not required

### CODE REVIEW

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      "Resource": "arn:aws:s3:::${var.ScanningBucketName}/*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListBucket"
+      ],
+      "Resource": "arn:aws:s3:::${var.ScanningBucketName}"
+    },
+    {
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:25]: Please follow the conditions from the original AWS template to create the policy.
https://github.com/trendmicro/cloudone-filestorage-plugins/blob/0c1e219478b3e8c2ff6aaf762a65847c0c3a9457/post-scan-actions/aws-python-promote-or-quarantine/template.yml#L72

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  type = string
+  description = "The method by which files were quarantined. (Options: move, copy)"
+  default = "move"
+  validation {
+    condition = contains(["move", "copy"], var.QuarantineMode)
+    error_message = "The quarantine mode must be either move or copy."
+  }
+ }
+ 
+variable "ACL" {
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:56]: Validate the value of  `ACL` by the options.

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Zip the lambda code
+data "archive_file" "zip_the_python_code" {
+type        = "zip"
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:3]:
```suggestion
  type        = "zip"
  source_file = "../handler.py"
  output_path = "../handler.zip"
```

File: post-scan-actions/aws-python-slack-notification/terraform/lambda.tf [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Zip the lambda code
+data "archive_file" "zip_the_python_code" {
+type        = "zip"
```
Comment (kwan-su) [inline post-scan-actions/aws-python-slack-notification/terraform/lambda.tf:3]:
```suggestion
  type        = "zip"
  source_file = "../handler.py"
  output_path = "${path.module}/lambda_zip/handler.zip"
```

**Comment by kwan-su** [comment]
Why this PR include the Terraform template for aws-python-slack-notification?

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  filename      = "../handler.zip"
+  function_name = "PromoteOrQuarantineLambda"
+  role          = aws_iam_role.PromoteOrQuarantineLambdaRole.arn
+  handler       = "handler.lambda_handler"
+  memory_size = 512
+  timeout = 30
+  runtime = "python3.8"
+  tracing_config {
+      mode = "Active"
+    }
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:19]:
```suggestion
  }
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+}
+EOF
+}
+
+# Attach the policy ScanningBucketNamePolicy to the role
+resource "aws_iam_role_policy_attachment" "ScanningBucketNamePolicyAttachment" {
+  depends_on = [aws_iam_role.PromoteOrQuarantineLambdaRole]
+  role = aws_iam_role.PromoteOrQuarantineLambdaRole.name
+  policy_arn = aws_iam_policy.ScanningBucketNamePolicy.arn
+}
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:68]: Add newline in the end of files.

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    source_arn = "${var.ScanResultTopicARN}"
+}
+
+# Create the sns event source mapping to lambda
+resource "aws_sns_topic_subscription" "ScanResult" {
+  depends_on = [aws_lambda_function.PromoteOrQuarantineLambda]
+  topic_arn = var.ScanResultTopicARN
+  protocol  = "lambda"
+  endpoint  = aws_lambda_function.PromoteOrQuarantineLambda.arn
+}
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:46]: Add newline.

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/outputs.tf [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# Outputs
+output "ScanResult" {
+   value = aws_lambda_function.PromoteOrQuarantineLambda.arn
+}
+
+output "PromoteOrQuarantineLambda" {
+   value = aws_lambda_function.PromoteOrQuarantineLambda.arn
+}
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/outputs.tf:8]: Add newline.

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/providers.tf [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+      source  = "hashicorp/aws"
+      version = "~> 4.22.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = var.region
+}
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/providers.tf:14]: Add newline.

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+  type = string
+  default = ""
+  description = "[Optional] Apply an access control list (ACL) on the file after it has been promoted or quarantined. (Options: private, public-read, public-read-write, authenticated-read, aws-exec-read, bucket-owner-read, bucket-owner-full-control)"
+ }
+ 
+ 
+
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:63]: Add newline.

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+  description = "The ARN of the scan result SNS topic in storage stack."
+ }
+
+variable "PromoteBucketName" {
+  type = string
+  default = ""
+  description = "[Optional] The bucket name used to promote files without malicious intent. Leave the bucket name blank to disable promoting."
+ }
+
+variable "PromoteMode" {
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:30]: Please follow the naming convention like we did before in GCP plugin's Terraform template for consistency. Apply the convention to the other variables too.
https://github.com/trendmicro/cloudone-filestorage-plugins/blob/0c1e219478b3e8c2ff6aaf762a65847c0c3a9457/post-scan-actions/gcp-python-promote-or-quarantine/variables.tf#L48
```suggestion
variable "promote_mode" {
```

File: post-scan-actions/aws-python-promote-or-quarantine/README.md [hunk 12]
```diff
@@ -26,6 +26,19 @@ After a scan occurs, this example Lambda function places clean files in one buck
 1. Visit [the app's page on the AWS Lambda Console](https://console.aws.amazon.com/lambda/home?#/create/app?applicationId=arn:aws:serverlessrepo:us-east-1:415485722356:applications/cloudone-filestorage-plugin-action-promote-or-quarantine).
 2. Click the `Copy as SAM Resource` button and paste the copied YAML into your SAM template, filling in any required parameters.
 
+### Terraform
+
+Make sure that you have terraform and AWS CLI installed and configured, then go to terraform folder and change appropriately the values at the variables in the `variables.tf` file.
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/README.md:31]:
```suggestion
Make sure that you have Terraform and AWS CLI installed and configured, then go to terraform folder and change appropriately the values at the variables in the `variables.tf` file.
```

**Comment by Dave-Lin-FSS** [comment]
Success Move / Copy to Promote / Quarantine by terraform template

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "s3:PutObjectTagging",
+        "s3:PutObjectAcl"
+      ],
+      "Resource": [
+        "arn:aws:s3:::${var.PromoteBucketName}/*",
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:33]: Please use the same condition from the original policy. 
https://github.com/trendmicro/cloudone-filestorage-plugins/blob/0c1e219478b3e8c2ff6aaf762a65847c0c3a9457/post-scan-actions/aws-python-promote-or-quarantine/template.yml#L122

Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:33]:
```suggestion
        %{ if local.promote_enalbed }"arn:aws:s3:::${var.PromoteBucketName}/*"%{ else }%{ endif }
        %{ if local.promote_enalbed && local.quaratine_enalbed },%{ else }%{ endif }
        %{ if local.quaratine_enalbed }"arn:aws:s3:::${var.QuarantineBucketName}/*"%{ else }%{ endif }
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        "arn:aws:s3:::${var.QuarantineBucketName}/*"
+      ]
+    }
+  ]
+}
+EOF
+}
+
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:41]:
```suggestion
locals {
  promote_enalbed = var.PromoteBucketName == "" ? false : true
  quaratine_enalbed = var.QuarantineBucketName == "" ? false : true
}
```

**Review by Dave-Lin-FSS — COMMENTED** [review]
20221114

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf (hunk 13 above)
Comment (Dave-Lin-FSS) [reply post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:33]:
```
# local variables for ScanningBucketNamePolicy Resource
locals {
  promote_enalbed = var.PromoteBucketName == "" ? false : true
  quaratine_enalbed = var.QuarantineBucketName == "" ? false : true
}
```

Comment (felipecosta09) [reply post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:33]: As far I know, these conditions only works on CF, if they actually work on terraform I will be happy to add them here. Could you confirm?

**Comment by felipecosta09** [comment]
> 20221114

?

**Comment by felipecosta09** [comment]
> If Terraform variables name are changed, please also modify the old variables with the new naming in the other files, and apply the deployment to ensure the template is working fine.

I will leave this one for the last until all the changes are made to avoid re-work

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf (hunk 2 above)
Comment (felipecosta09) [reply post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf]: That has been done by one of the commit suggestions.

**Comment by felipecosta09** [comment]
> Why this PR include the Terraform template for aws-python-slack-notification?

That was a mistake, I will fix it...

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf (hunk 10 above)
Comment (felipecosta09) [reply post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf]: Done

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/providers.tf (hunk 9 above)
Comment (felipecosta09) [reply post-scan-actions/aws-python-promote-or-quarantine/terraform/providers.tf]: Done

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/outputs.tf (hunk 8 above)
Comment (felipecosta09) [reply post-scan-actions/aws-python-promote-or-quarantine/terraform/outputs.tf]: Done

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf (hunk 6 above)
Comment (felipecosta09) [reply post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf]: Done

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+  type = string
+  description = "AWS region"
+  default = "us-east-1"
+  validation {
+    condition     = can(regex("[a-z][a-z]-[a-z]+-[1-2]", var.region))
+    error_message = "Must be valid AWS Region names."
+  }
+ }
+
+variable "ScanningBucketName" {
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:12]:
```suggestion
variable "scanning_bucket_name" {
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+  }
+ }
+
+variable "quarantine_bucket_name" {
+  type = string
+  default = ""
+  description = "[Optional] The bucket name to quarantine malicious files. Leave the bucket name blank to disable quarantining"
+ }
+
+ variable "QuarantineMode" {
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:46]:
```suggestion
 variable "quarantine_mode" {
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 17]
```diff
@@ -0,0 +1,48 @@
+# Zip the lambda code
+data "archive_file" "zip_the_python_code" {
+  type        = "zip"
+  source_file = "../handler.py"
+  output_path = "../handler.zip"
+source_file = "../handler.py"
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:6]:
```suggestion
  source_file = "../handler.py"
  output_path = "../handler.zip"
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 18]
```diff
@@ -0,0 +1,70 @@
+# Lambda execution policy for ScanningBucketNamePolicy
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:1]:
```suggestion
# Lambda execution policy for ScanningBucketNamePolicy
locals {
  promote_enalbed = var.promote_bucket_name == "" ? false : true
  quaratine_enalbed = var.quarantine_bucket_name == "" ? false : true
}
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+locals {
+  promote_enalbed = var.promote_bucket_name == "" ? false : true
+  quaratine_enalbed = var.quarantine_bucket_name == "" ? false : true
+}
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:5]:
```suggestion
}

```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 20]
```diff
@@ -0,0 +1,49 @@
+# Zip the lambda code
+data "archive_file" "zip_the_python_code" {
+  type        = "zip"
+  source_file = "../handler.py"
+  output_path = "../handler.zip"
+  source_file = "../handler.py"
+  output_path = "../handler.zip"
+output_path = "../handler.zip"
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:8]:
```suggestion

```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 21]
```diff
@@ -0,0 +1,49 @@
+# Zip the lambda code
+data "archive_file" "zip_the_python_code" {
+  type        = "zip"
+  source_file = "../handler.py"
+  output_path = "../handler.zip"
+  source_file = "../handler.py"
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:6]:
```suggestion

```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetObject",
+        "s3:DeleteObject",
+        "s3:GetObjectTagging"
+      ],
+      "Resource": "arn:aws:s3:::${var.ScanningBucketName}/*"
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:20]:
```suggestion
      "Resource": "arn:aws:s3:::${var.scanning_bucket_name}/*"
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListBucket"
+      ],
+      "Resource": "arn:aws:s3:::${var.ScanningBucketName}"
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:27]:
```suggestion
      "Resource": "arn:aws:s3:::${var.scanning_bucket_name}"
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "s3:PutObjectTagging",
+        "s3:PutObjectAcl"
+      ],
+      "Resource": [
+        %{ if local.promote_enalbed }"arn:aws:s3:::${var.PromoteBucketName}/*"%{ else }%{ endif }
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:37]:
```suggestion
        %{ if local.promote_enalbed }"arn:aws:s3:::${var.promote_bucket_name}/*"%{ else }%{ endif }
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 25]
```diff
@@ -0,0 +1,74 @@
...
+        %{ if local.promote_enalbed && local.quaratine_enalbed },%{ else }%{ endif }
+        %{ if local.quaratine_enalbed }"arn:aws:s3:::${var.QuarantineBucketName}/*"%{ else }%{ endif }
```
Comment (kwan-su) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:39]:
```suggestion
        %{ if local.quaratine_enalbed }"arn:aws:s3:::${var.quarantine_bucket_name}/*"%{ else }%{ endif }
```

**Comment by kwan-su** [comment]
> > Why this PR include the Terraform template for aws-python-slack-notification?
> 
> That should be fixed as well...

The folder "aws-python-slack-notification" still in this PR.
[post-scan-actions/aws-python-slack-notification](https://github.com/trendmicro/cloudone-filestorage-plugins/pull/101/files#diff-20c5f7067f713ecc91c224243f1c9e2b8be897a14d7f32e2fb2bf9ee188c31a4)

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 26]
```diff
@@ -0,0 +1,74 @@
...
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetObject",
+        "s3:DeleteObject",
+        "s3:GetObjectTagging"
+      ],
+      "Resource": "arn:aws:s3:::${var.ScanningBucketName}/*"
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:21]:
```suggestion
      "Resource": "arn:aws:s3:::${var.scanning_bucket_name}/*"
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:ListBucket"
+      ],
+      "Resource": "arn:aws:s3:::${var.ScanningBucketName}"
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:28]:
```suggestion
      "Resource": "arn:aws:s3:::${var.scanning_bucket_name}"
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 28]
```diff
@@ -0,0 +1,74 @@
...
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:PutObject",
+        "s3:PutObjectTagging",
+        "s3:PutObjectAcl"
+      ],
+      "Resource": [
+        %{ if local.promote_enalbed }"arn:aws:s3:::${var.PromoteBucketName}/*"%{ else }%{ endif }
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:38]:
```suggestion
        %{ if local.promote_enalbed }"arn:aws:s3:::${var.promote_bucket_name}/*"%{ else }%{ endif }
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 29]
```diff
@@ -0,0 +1,74 @@
...
+        %{ if local.promote_enalbed && local.quaratine_enalbed },%{ else }%{ endif }
+        %{ if local.quaratine_enalbed }"arn:aws:s3:::${var.QuarantineBucketName}/*"%{ else }%{ endif }
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:40]:
```suggestion
        %{ if local.quaratine_enalbed }"arn:aws:s3:::${var.quarantine_bucket_name}/*"%{ else }%{ endif }
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf [hunk 30]
```diff
@@ -0,0 +1,74 @@
...
+ ]
+}
+EOF
+}
+
+# Attach the policy ScanningBucketNamePolicy to the role
+resource "aws_iam_role_policy_attachment" "ScanningBucketNamePolicyAttachment" {
+  depends_on = [aws_iam_role.PromoteOrQuarantineLambdaRole]
+  role = aws_iam_role.PromoteOrQuarantineLambdaRole.name
+  policy_arn = aws_iam_policy.ScanningBucketNamePolicy.arn
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/iam.tf:73]:
```suggestion
  policy_arn = aws_iam_policy.scanning_bucket_name_policy.arn
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 31]
```diff
@@ -0,0 +1,47 @@
+# Zip the lambda code
+data "archive_file" "zip_the_python_code" {
+  type        = "zip"
+  source_file = "../handler.py"
+  output_path = "../handler.zip"
+  output_path = "../handler.zip"
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:6]:
```suggestion
  
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
+  handler       = "handler.lambda_handler"
+  memory_size = 512
+  timeout = 30
+  runtime = "python3.8"
+  tracing_config {
+      mode = "Active"
+  }
+  environment {
+    variables = {
+      PROMOTEBUCKET= var.PromoteBucketName
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:23]:
```suggestion
      PROMOTEBUCKET= var.promote_bucket_name
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 33]
```diff
@@ -0,0 +1,47 @@
...
+      PROMOTEMODE= var.PromoteMode
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:24]:
```suggestion
      PROMOTEMODE= var.promote_mode
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 34]
```diff
@@ -0,0 +1,47 @@
...
+      QUARANTINEBUCKET= var.QuarantineBucketName
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:25]:
```suggestion
      QUARANTINEBUCKET= var.quarantine_bucket_name
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 35]
```diff
@@ -0,0 +1,47 @@
...
+      QUARANTINEMODE= var.QuarantineMode
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:26]:
```suggestion
      QUARANTINEMODE= var.quarantine_mode
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 36]
```diff
@@ -0,0 +1,47 @@
...
+      ACL= var.ACL
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:27]:
```suggestion
      ACL= var.acl
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 37]
```diff
@@ -0,0 +1,47 @@
...
+  }
+}
+
+# Allows Lambda to add a trigger to SNS
+resource "aws_lambda_permission" "PromoteOrQuarantineLambdaPermission" {
+    statement_id = "AllowExecutionFromSNS"
+    action = "lambda:InvokeFunction"
+    function_name = "${aws_lambda_function.PromoteOrQuarantineLambda.arn}"
+    principal = "sns.amazonaws.com"
+    source_arn = "${var.ScanResultTopicARN}"
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:38]:
```suggestion
    source_arn = "${var.scan_result_topic_arn}"
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 38]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+# Create the sns event source mapping to lambda
+resource "aws_sns_topic_subscription" "ScanResult" {
+  depends_on = [aws_lambda_function.PromoteOrQuarantineLambda]
+  topic_arn = var.ScanResultTopicARN
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:44]:
```suggestion
  topic_arn = var.scan_result_topic_arn
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 39]
```diff
@@ -0,0 +1,64 @@
...
+  default = ""
+  description = "[Optional] The bucket name used to promote files without malicious intent. Leave the bucket name blank to disable promoting."
+ }
+
+variable "promote_mode" {
+  type = string
+  description = "The method by which files were promoted. (Options: move, copy)"
+  default = "move"
+  validation {
+    condition = contains(["move", "copy"], var.PromoteMode)
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:35]:
```suggestion
    condition = contains(["move", "copy"], var.promote_mode)
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 40]
```diff
@@ -0,0 +1,64 @@
...
+  default = ""
+  description = "[Optional] The bucket name to quarantine malicious files. Leave the bucket name blank to disable quarantining"
+ }
+
+ variable "quarantine_mode" {
+  type = string
+  description = "The method by which files were quarantined. (Options: move, copy)"
+  default = "move"
+  validation {
+    condition = contains(["move", "copy"], var.QuarantineMode)
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:51]:
```suggestion
    condition = contains(["move", "copy"], var.quarantine_mode)
```

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf [hunk 41]
```diff
@@ -0,0 +1,64 @@
...
+    error_message = "The quarantine mode must be either move or copy."
+  }
+ }
+
+variable "acl" {
+  type = string
+  default = ""
+  description = "[Optional] Apply an access control list (ACL) on the file after it has been promoted or quarantined. (Options: private, public-read, public-read-write, authenticated-read, aws-exec-read, bucket-owner-read, bucket-owner-full-control)"
+  validation {
+    condition = contains(["private, public-read", "public-read-write", "authenticated-read", "aws-exec-read", "bucket-owner-read", "bucket-owner-full-control"], var.acl)
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/variables.tf:61]:
```suggestion
    condition = contains(["", "private", "public-read", "public-read-write", "authenticated-read", "aws-exec-read", "bucket-owner-read", "bucket-owner-full-control"], var.acl)
```

**Review by Dave-Lin-FSS — COMMENTED** [review]
20221124
Correcting parameter naming and terraform apply success

**Comment by kwan-su** [comment]
> > > Why this PR include the Terraform template for aws-python-slack-notification?
> > 
> > 
> > That should be fixed as well...
> 
> The folder "aws-python-slack-notification" still in this PR. [post-scan-actions/aws-python-slack-notification](https://github.com/trendmicro/cloudone-filestorage-plugins/pull/101/files#diff-20c5f7067f713ecc91c224243f1c9e2b8be897a14d7f32e2fb2bf9ee188c31a4)

Please remove the changes for  `aws-python-slack-notification` in this PR.

File: post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf [hunk 42]
```diff
@@ -0,0 +1,47 @@
+# Zip the lambda code
+data "archive_file" "zip_the_python_code" {
+  type        = "zip"
+  source_file = "../handler.py"
+  output_path = "../handler.zip"
+
```
Comment (Dave-Lin-FSS) [inline post-scan-actions/aws-python-promote-or-quarantine/terraform/lambda.tf:6]: Suggest remove unused empty line.

**Comment by felipecosta09** [comment]
All the references from others files in the Slack plugin should be removed now.

**Comment by felipecosta09** [comment]
Hey @jack-c-tang can you review this PR?

**Review by Dave-Lin-FSS — APPROVED** [review]
LGTM

**Comment by kwan-su** [comment]
Hi @felipecosta09 , Please squash your commits.

**Comment by felipecosta09** [comment]
> Hi @felipecosta09 , Please squash your commits.

Hi @kwan-su please squash from your side, it is just a button click. Check the #94

**Review by kwan-su — APPROVED** [review]

**Pull request merged**